xml/security_ssh.xml

```diff
@@ -550,40 +530,151 @@ cd path                            Change remote directory to 'path'
   </note>
 
   <sect2 xml:id="sec-ssh-authentic-gen-key">
-   <title>Generating an SSH key</title>
-   <procedure>
-    <step>
+   <title>Generating SSH keys</title>
+   <para>
+     There are several key types to choose from: DSA, RSA, ECDSA, ECDSA-SK,
+     Ed25519, and Ed25519-SK. DSA was deprecated in OpenSSH 7.0 and should
+     not be used. RSA is the most universal and is accepted by older clients.
+     Ed25519 and ECDSA are stronger, faster, and smaller. Ed25519 is
+     considered to be the strongest. If you must support older clients
+     that do not support Ed25519, create both Ed25519 and RSA host keys,
+     and clients will automatically use the strongest key they support.
+   </para>
+   <note>
+     <title>Older clients may be unsafe</title>
      <para>
-      To generate a key with default parameters (RSA, 2048 bits), enter the
-      command <command>ssh-keygen</command>.
+       Some older SSH clients do not support ECDSA and ED25519. ECDSA and
+       ED25519 were released with OpenSSH 6.5 in 2014. It is
+       important to keep security services updated, and, if possible,
+       to not allow unsafe old clients.
      </para>
-    </step>
+   </note>
+   <para>
+     SSH keys serve two purposes: authenticating servers with host keys, and
+     authenticating clients. Host keys are stored in
+     <filename>/etc/ssh</filename>. Client keys are specific to users, and
+     are stored in <filename>/home/<replaceable>user</replaceable>/.ssh</filename>.
+   </para>
+   <para>
+     Host keys must not have passphrases.
```

Comment:
a passphrase. Don't think you can have multiple passphrases

Reply:
"keys" is plural, so "passphrases" should be plural. But this gets confused all the time, like "Host keys must not have a passphrase", which is not grammatically correct. I changed the wording to make it clearer.